### ISSUE TITLE
[Docs] Usage of gpytorch.settings.use_toeplitz in SKIP GPs

### ISSUE THREAD

**Issue by activatedgeek** [cornellius-gp/gpytorch#1265]
# 📚 Documentation/Examples

Example in question: https://docs.gpytorch.ai/en/v1.1.1/examples/02_Scalable_Exact_GPs/Scalable_Kernel_Interpolation_for_Products_CUDA.html

In code cell 7, `train` is called with 

```python
...
with gpytorch.settings.use_toeplitz(True):
    %time train()
...
```

Later, however, inside `train` this same context is set to `False`.

```python
...
with gpytorch.settings.use_toeplitz(False), gpytorch.settings.max_root_decomposition_size(30):
    # Get output from model
    output = model(train_x)
...
```

Two questions:
- The call to `optimizer.step()` remains out of this context. Is this intentional? Does it matter?
- Doesn't setting this to `False` again undo the previous context? As I understand, the kernel we are using for grid interpolation should be utilizing Toeplitz structure?

Further, I also have this warning which seems to concern with usage of `nonzero()` in PyTorch.

```shell
lib/python3.7/site-packages/gpytorch/utils/interpolation.py:119: UserWarning: This overload of nonzero is deprecated:
        nonzero()
Consider using one of the following signatures instead:
        nonzero(*, bool as_tuple) (Triggered internally at  /pytorch/torch/csrc/utils/python_arg_parser.cpp:766.)
  left_boundary_pts = (lower_grid_pt_idxs < 0).nonzero()
```

Thanks!

## Versions:

**GPyTorch**: 1.1.1
**PyTorch**: 1.6.0

**Comment by KeAWang** [comment]
I think this is set to false because using Toeplitz structure is slow on a GPU unless you have a ton of inducing points. It's on by default, but this notebook is showcasing performance with CUDA, which is likely why the setting is turned off.

**Comment by activatedgeek** [comment]
Thanks!

I did see the [docs](https://docs.gpytorch.ai/en/v1.1.1/settings.html#gpytorch.settings.use_toeplitz) mention this

> Whether or not to use Toeplitz math with gridded data, grid inducing point modules Pros: memory efficient, faster on CPU Cons: slower on GPUs with < 10000 inducing points

Any ideas if the `optimizer.step()` call has anything to do with this? My guess would be no but just wanted to confirm.

**Comment by KeAWang** [comment]
No it's not because of the optimizer

**Comment by activatedgeek** [comment]
No, I meant whether the optimizer call is within context or outside, doesn't matter. Is this correct?

**Comment by KeAWang** [comment]
No `optimizer.step()` will just update the parameters by the gradients so it isn't affected
